flink-end-to-end-tests/flink-bucketing-sink-test/pom.xml

```diff
@@ -72,30 +66,15 @@
 			<plugin>
 				<groupId>org.apache.maven.plugins</groupId>
 				<artifactId>maven-shade-plugin</artifactId>
-				<version>3.0.0</version>
 				<executions>
 					<execution>
+						<id>BucketingSinkTestProgram</id>
```

Comment:
I am not sure what value it brings, but I guess it doesn't bother.

Reply:
There are multiple benefits to introducing an explicit id.

They reduce the chance for clashes between executions, as executions without IDs are treated as a single execution.
The ID is printed when the shade-plugin is executed, which makes it easier to associate the output to a specific shade-plugin execution.
Finally, with newer maven versions you can invoke specific executions.

Reply:
Agree, but as in most modules there is a single execution for `maven-shade-plugin` points 1 and 3 are irrelevant, right? The only module where there are multiple executions is queryable-state, I think. But as I said it doesn't bother, just wanted to ask for my understanding.

Reply:
point 1 isn't irrelevant in terms of maintainability; without an execution ID it could happen fairly easily that a shade-plugin configuration in any of the parent modules could modify the behavior accidentally.
The last point is also still relevant as there are in fact 2 `shade-plugin` executions defined here, `shade-flink` inherited from the root pom and the one (or two) in the actual module.

Reply:
Ok point taken, thanks for explaining it to me.